app/routes/offers/index.js

```diff
@@ -1,34 +1,38 @@
-import AuthorizeRoute from './../authorize';
+import AuthorizeRoute from "./../authorize";
 
 export default AuthorizeRoute.extend({
-
   model() {
-    return this.store.peekAll('offer');
+    let cachedRecords = this.store.peekAll("offer");
+    if (cachedRecords.get("length")) {
+      return Ember.RSVP.resolve(cachedRecords);
```

Comment:
Since you are using `Ember.RSVP`, giving error `Ember is not defined` as it's not imported yet.

Reply:
Sorry, It was added by mistake, I have removed it.